### PR TITLE
[FR] DRAFT Release Workflow on PR Merge

### PR DESCRIPTION
**Pull request by Mikaayenson**
# Pull Request

*Issue link(s)*:

## Summary - What I changed

- We need to make sure the drafts are updated based on the context on the PR so that the labels influence the tagging.
- Simplified the workflow to update the draft when the PR merges 

## Checklist

- [x] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `maintenance`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [x] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [ ] Secret and sensitive material has been managed correctly
- [ ] Automated testing was updated or added to match the most common scenarios
- [ ] Documentation and comments were added for features that require explanation

### CODE REVIEW

File: .github/workflows/version-code-and-release.yml [hunk 1]
```diff
@@ -11,12 +11,7 @@ on:
       - 'detection_rules/**'
       - 'tests/**'
       - '**/*.md'
-    types: [opened, reopened, synchronize, labeled]
-  push:
-    branches:
-      - main
-    paths:
-      - 'pyproject.toml'
+    types: [opened, reopened, synchronize, labeled, closed]
```
Comment (eric-forte-elastic) [inline .github/workflows/version-code-and-release.yml:14]:
```suggestion
    types: [opened, reopened, synchronize, labeled]
```

Just double checking, are we sure we want closed?

Comment (eric-forte-elastic) [reply .github/workflows/version-code-and-release.yml:14]: Oh wait, we do want closed because we would want this check to run after a merge too right?

Comment (Mikaayenson) [reply .github/workflows/version-code-and-release.yml:14]: yes thats for the last step.

**Comment by protectionsmachine** [comment]
## Bug - Guidelines

These guidelines serve as a reminder set of considerations when addressing a bug in the code.

### Documentation and Context

- [ ] Provide detailed documentation (description, screenshots, reproducing the bug, etc.) of the bug if not already documented in an issue.
- [ ] Include additional context or details about the problem.
- [ ] Ensure the fix includes necessary updates to the release documentation and versioning.

### Code Standards and Practices

- [ ] Code follows established design patterns within the repo and avoids duplication.
- [ ] Code changes do not introduce new warnings or errors.
- [ ] Variables and functions are well-named and descriptive.
- [ ] Any unnecessary / commented-out code is removed.
- [ ] Ensure that the code is modular and reusable where applicable.
- [ ] Check for proper exception handling and messaging.

### Testing

- [ ] New unit tests have been added to cover the bug fix or edge cases.
- [ ] Existing unit tests have been updated to reflect the changes.
- [ ] Provide evidence of testing and detecting the bug fix (e.g., test logs, screenshots).
- [ ] Validate that any rules affected by the bug are correctly updated.
- [ ] Ensure that performance is not negatively impacted by the changes.
- [ ] Verify that any release artifacts are properly generated and tested.

### Additional Checks

- [ ] Ensure that the bug fix does not break existing functionality.
- [ ] Review the bug fix with a peer or team member for additional insights.
- [ ] Verify that the bug fix works across all relevant environments (e.g., different OS versions).
- [ ] Confirm that all dependencies are up-to-date and compatible with the changes.
- [ ] Confirm that the proper version label is applied to the PR `patch`, `minor`, `major`.

**Review by eric-forte-elastic — APPROVED** [review]
:green_circle: Manual review, looks good to me! :+1:

**Review by terrancedejesus — APPROVED** [review]
